c#/Mandoline.Api.Examples/Core/Domain/Download.cs

```diff
@@ -4,116 +4,112 @@
 
-        // runs a download and returns a list of dataseries lists, effectively preserving
-        // the pages as they arrived
-        private static async Task<List<List<DataseriesDto>>> DownloadPages()
+        // set up download request
+        ControllerDownloadRequestDto req = new ControllerDownloadRequestDto()
         {
-            // get our sample selection
-            SelectionDto sampleSelect = AppConstants.SampleSelection.GetInstance();
+            selections = new SelectionDto[] { sampleSelect },
+            format = FileFormat.Csv,
+            name = "SampleFileDownload-" + DateTime.Now.ToString("yyyyMMddHHmm") + ".csv",
+        };
+
+        // make file request
+        Client.Models.Assertion<string> fileResult = await api.DownloadFileAsync(
+            req,
+            new System.Threading.CancellationTokenSource(TimeSpan.FromMinutes(5)).Token).ConfigureAwait(true);
+        output.PrintData(fileResult.Result);
+    }
 
-            // set the number of items (i.e. in this case DataseriesDto objects) to be included in each page of data
-            const int PAGE_SIZE = 5;
+    // downloads a sample selection from the macro databank, loading up onto the DataGridView
+    public static async Task RunDownloadAsync(Output output)
+    {
+        // queue api requests till all pages of data have been received
+        // first section here returns a list of data series objects
+        List<List<DataseriesDto>> downloadResult = await DownloadPages().ConfigureAwait(true);
+        output.PrintData(downloadResult);
+    }
 
-            // initialize api object
-            var api = new ApiClient(AppConstants.BaseURL, AppConstants.ApiToken);
+    // runs a download and returns a list of dataseries lists, effectively preserving
+    // the pages as they arrived
+    private static async Task<List<List<DataseriesDto>>> DownloadPages()
+    {
+        // get our sample selection
+        SelectionDto sampleSelect = AppConstants.SampleSelection.GetInstance();
 
-            // this will track which page of data is currently being requested from the api
-            int page = 0;
-            List<DataseriesDto> newPage;
-            var pageList = new List<List<DataseriesDto>>();
+        // set the number of items (i.e. in this case DataseriesDto objects) to be included in each page of data
+        const int PAGE_SIZE = 5;
 
-            // loop till the number of elements in the new page of data doesn't equal the page size we expect
-            do
-            {
-                // make the download request
-                var downloadResult = await api.DownloadAsync(
-                    sampleSelect,
-                    new System.Threading.CancellationTokenSource(TimeSpan.FromMinutes(5)).Token,
-                    page,
-                    PAGE_SIZE).ConfigureAwait(true);
+        // initialize api object
+        ApiClient api = new ApiClient(AppConstants.BaseURL, AppConstants.ApiToken);
```

Comment:
here as well, it feels like the code is being repeated when someone reads it but ok it is just a personal preference tbh

Reply:
Yeah I agree with @asfondilis  why not have apiclient as a property initialized in the constructor? Either by DI or just manually.

Reply:
I didn't write this code. However, I think it is because they are static methods that the ApiClient is being regenerated on every call. Perhaps I should make them non-static and implement the constructor as suggested?